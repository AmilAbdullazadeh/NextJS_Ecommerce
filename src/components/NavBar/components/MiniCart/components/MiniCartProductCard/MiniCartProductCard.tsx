/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import RemoveIcon from '@material-ui/icons/RemoveOutlined';
import AddIcon from '@material-ui/icons/AddCircleOutlineOutlined';
import DeleteIcon from '@material-ui/icons/DeleteOutline';
import Button from '@material-ui/core/Button';

import { CartProduct } from '@core';
import { ImageWithFallback, Typography } from '@components';
import { useCart } from '@hooks';

export interface MiniCartProductCardProps {
    item: CartProduct;
}

//https://codepen.io/btcrooks/pen/MbNape

const MiniCartProductCard: React.FC<MiniCartProductCardProps> = ({
    item: { product, quantity },
}: MiniCartProductCardProps) => {
    const {
        actions: {
            decreaseCartInventory,
            increaseCartInventory,
            removeFromCart,
        },
    } = useCart();

    const handleDecrease = () =>
        decreaseCartInventory({
            productId: product.id,
        });

    const handleAdd = () =>
        increaseCartInventory({
            productId: product.id,
        });

    const handleRemove = () => removeFromCart({ productId: product.id });

    return (
        <article>
            <Box my={2} minWidth={300}>
                <Grid
                    container
                    direction="row"
                    justifyContent="flex-start"
                    alignItems="flex-start"
                >
                    <Grid item xs={4}>
                        <Box component="div" display="block">
                            <ImageWithFallback
                                src={product.image ?? ''}
                                fallbackSrc={'phoneFallBack.png'}
                                alt={product.name}
                                width={100}
                                height={100}
                            />
                        </Box>
                    </Grid>
                    <Grid item xs={8}>
                        <Typography display="inline">{product.name}</Typography>
                        <Typography display="inline" color="primary">
                            {` (x${quantity})`}
                        </Typography>
                        <Box mt={1}>
                            <Typography>{product.price}â‚¬</Typography>
                        </Box>
                        <Grid
                            container
                            direction="row"
                            justifyContent="space-between"
                            alignItems="center"
                        >
                            <Grid item xs={6}>
                                <Button
                                    variant="text"
                                    startIcon={<RemoveIcon color="error" />}
                                    onClick={handleDecrease}
                                >
                                    <Typography variant="body2">
                                        remove
                                    </Typography>
                                </Button>
                            </Grid>
                            <Grid item xs={6}>
                                <Button
                                    variant="text"
                                    startIcon={<AddIcon color="primary" />}
                                    onClick={handleAdd}
                                >
                                    <Typography variant="body2">add</Typography>
                                </Button>
                            </Grid>
                        </Grid>
                        <div>
                            <Button
                                variant="text"
                                startIcon={<DeleteIcon color="error" />}
                                onClick={handleRemove}
                            >
                                <Typography variant="body2">
                                    remove from cart
                                </Typography>
                            </Button>
                        </div>
                    </Grid>
                </Grid>
            </Box>
        </article>
    );
};

export default MiniCartProductCard;
