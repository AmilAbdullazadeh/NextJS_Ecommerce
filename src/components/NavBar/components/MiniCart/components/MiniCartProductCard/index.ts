export { default as MiniCartProductCard } from './MiniCartProductCard';
