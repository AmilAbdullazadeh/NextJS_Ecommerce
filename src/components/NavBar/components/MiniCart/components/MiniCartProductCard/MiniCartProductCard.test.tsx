import React from 'react';

import { mockState, renderWithProviders } from '@testUtils';

import MiniCartProductCard, {
    MiniCartProductCardProps,
} from './MiniCartProductCard';

describe('MiniCartProductCard', () => {
    const defaultProps: MiniCartProductCardProps = {
        item: mockState.cart.products[0],
    };

    it('matches snapshot', () => {
        const {
            container: { firstChild },
        } = renderWithProviders(<MiniCartProductCard {...defaultProps} />);

        expect(firstChild).toMatchSnapshot();
    });
});
