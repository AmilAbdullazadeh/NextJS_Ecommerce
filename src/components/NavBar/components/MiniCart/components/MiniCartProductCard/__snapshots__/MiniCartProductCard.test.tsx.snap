// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniCartProductCard matches snapshot 1`] = `
<article>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          class="MuiBox-root MuiBox-root-2"
        >
          <div
            style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="display: block; box-sizing: border-box; padding-top: 100%;"
            />
            <noscript />
            <img
              alt="mock product name"
              data-nimg="true"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-displayInline"
        >
          mock product name
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-displayInline"
        >
           (x1)
        </p>
        <div
          class="MuiBox-root MuiBox-root-3"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            50
            â‚¬
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 13H5v-2h14v2z"
                    />
                  </svg>
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  remove
                </p>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body2"
                >
                  add
                </p>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorError"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                remove from cart
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</article>
`;
